### PR TITLE
Redis password

### PR DESCRIPTION
**Pull request by jayminkapish**
PR for https://github.com/overview/overview-local/issues/14

I now have this running in NYT INT Kubernetes cluster without problems. I recommend please test this against a redis instance with Auth disabled. I'm not sure if RedisClient library will ignore the password if its empty.

Looking forward to your feedback.

### CODE REVIEW

File: web/app/modules/RedisModule.scala [hunk 1]
```diff
@@ -8,14 +8,14 @@ import play.api.Configuration
...
+case class RedisConfiguration(host: String, port: Int, password: String)
 
 /** Provides a RedisClient. */
 @Singleton
 class RedisModule @Inject() (actorSystem: ActorSystem, configuration: Configuration, lifecycle: ApplicationLifecycle) {
   lazy val client: RedisClient = {
-    val config = RedisConfiguration(configuration.get[String]("redis.host"), configuration.get[Int]("redis.port"))
-    val ret = new RedisClient(config.host, config.port)(actorSystem)
+    val config = RedisConfiguration(configuration.get[String]("redis.host"), configuration.get[Int]("redis.port"), password=configuration.get[String]("redis.password"))
+    val ret = new RedisClient(config.host, config.port, password=Option(config.password))(actorSystem)
```
Comment (adamhooper) [inline web/app/modules/RedisModule.scala]: er ... this compiles?

Comment (adamhooper) [reply web/app/modules/RedisModule.scala]: Oh, I see. Better question: does this work if you run `./dev`?

Comment (adamhooper) [reply web/app/modules/RedisModule.scala]: Ah, https://github.com/antirez/redis/issues/3676 -- empty password means no auth.

Comment (jayminkapish) [reply web/app/modules/RedisModule.scala]: Yep this compiles on my end. Are you getting an error? I haven't tried `./dev`. I ran those `./sbt` and docker/build commands you gave me, built `overview-web` image, pushed to our GCP project and using it in the Kube deployment.

**Comment by adamhooper** [comment]
You got it totally right. It does indeed work everywhere, because Redis treats `AUTH ""` (the default) as "no authentication". I added a couple of comments to make that clear.

**Comment by adamhooper** [comment]
Thank you for the pull request and testing. It's exciting that you can run it in NYT k8s -- that means you have a whole world of forks and bugfixes and whatnot open to you should you need it.

Now that we've merged your pull request, Jenkins will pick it up at https://jenkins-ci.overviewdocs.com/job/overview-server/. Assuming it succeeds, you'll be able to pull `overview/overview-web:0b720d8672ee371d853d48bdd5986957b4cada21` from Docker Hub. (Sometimes things fail, because we still haven't hunted down the last remaining races from our integration tests. If it fails, I'll ping Jenkins until it succeeds.)

**Comment by jayminkapish** [comment]
Thanks Adam for your quick response. Let me know if I can help in any way.

**Pull request merged**